### ISSUE TITLE
Magicgui 0.2.7 update broke affinder — missing required argument despite binding

### ISSUE THREAD

**Issue by jni** [napari/magicgui#179]
I still need to investigate this one, but thought I'd raise the issue first in case it's obvious to you @tlambert03. Trying affinder now raises the following error:

```python
Traceback (most recent call last):
  File "c:\users\demar\.conda\envs\napari-env\lib\site-packages\magicgui\widgets\_bases\value_widget.py", line 44, in <lambda>
    lambda *x: self.changed(value=x[0] if x else None)
  File "c:\users\demar\.conda\envs\napari-env\lib\site-packages\magicgui\events.py", line 603, in __call__
    self._invoke_callback(cb, event)
  File "c:\users\demar\.conda\envs\napari-env\lib\site-packages\magicgui\events.py", line 625, in _invoke_callback
    cb_event=(cb, event),
  File "c:\users\demar\.conda\envs\napari-env\lib\site-packages\magicgui\events.py", line 619, in _invoke_callback
    cb(event)
  File "c:\users\demar\.conda\envs\napari-env\lib\site-packages\magicgui\widgets\_function_gui.py", line 184, in _disable_button_and_call
    self.__call__()
  File "c:\users\demar\.conda\envs\napari-env\lib\site-packages\magicgui\widgets\_function_gui.py", line 283, in __call__
    value = self._function(*bound.args, **bound.kwargs)
  File "c:\users\demar\.conda\envs\napari-env\lib\site-packages\affinder\affinder.py", line 147, in start_affinder
    close_affinder()
  File "c:\users\demar\.conda\envs\napari-env\lib\site-packages\magicgui\widgets\_function_gui.py", line 275, in __call__
    raise TypeError(msg) from None
TypeError: missing a required argument: 'layers' in call to 'close_affinder(layers, callback)'.
To avoid this error, you can bind a value or callback to the parameter:

    close_affinder.layers.bind(value)

Or use the 'bind' option in the magicgui decorator:

    @magicgui(layers={'bind': value})
    def close_affinder(layers, callback): ...
```

Of course I *do* call `close_affinder.layers.bind` [here](https://github.com/jni/affinder/blob/b538689a92ce878efae8316cb981fd0c9fb9ced8/affinder/affinder.py#L141-L142), so somehow that is not "sticking". Or, rather, it seems maybe close_affinder is now getting called at creation time, since this happens when clicking "start", ie before we've (manually) tried to call close_affinder.

**Comment by tlambert03** [comment]
arg :/ not immediately obvious ... will have a look

**Comment by jni** [comment]
After a pair session with @tlambert03, we determined #139 was the culprit, specifically, the setting of the text on the call button there, which interferes with affinder's own text-setting aspirations. 😂

**Comment by tlambert03** [comment]
# 🤦‍♂️

**Comment by jni** [comment]
Confirmed that this was fixed by #180. Thank you @tlambert03!
